extension/bearertokenauthextension/bearertokenauth_test.go

```diff
@@ -18,15 +18,19 @@ import (
 )
 
 func TestPerRPCAuth(t *testing.T) {
```

Comment:
I'm not confident enough on this test: it doesn't prove me that there's a bug with the current code, and thus, doesn't ensure that the bug won't happen again. Can you please add a test like the following one:

```go
func TestUpdateToken(t *testing.T) {
	// prepare
	cfg := createDefaultConfig().(*Config)
	cfg.BearerToken = "1234"

	bauth := newBearerTokenAuth(cfg, zaptest.NewLogger(t))
	assert.NotNil(t, bauth)

	perRPCAuth, err := bauth.PerRPCCredentials()
	assert.NoError(t, err)

	ctx := context.Background()
	assert.NoError(t, bauth.Start(ctx, componenttest.NewNopHost()))

	// initial token, OK
	_, err = bauth.Authenticate(ctx, map[string][]string{"authorization": {"Bearer " + "1234"}})
	assert.NoError(t, err)
	md, err := perRPCAuth.GetRequestMetadata(context.Background())
	assert.NoError(t, err)
	assert.Equal(t, map[string]string{"authorization": "Bearer " + "1234"}, md)

	// update the token
	bauth.setAuthorizationValue("5678")
	_, err = bauth.Authenticate(ctx, map[string][]string{"authorization": {"Bearer " + "5678"}})
	assert.NoError(t, err)

	md, err = perRPCAuth.GetRequestMetadata(context.Background())
	assert.NoError(t, err)
	assert.Equal(t, map[string]string{"authorization": "Bearer " + "5678"}, md)

	assert.NoError(t, bauth.Shutdown(context.Background()))
}
```

Reply:
Of course I could add the test.

Regarding if it's an issue, `PerRPCCredentials()` would be called for only once when one exporter connects to the server endpoints so the token inside metadata is static and won't get a chance to update.

Reply:
Added the test while I removed calls to `Authenticate()` as the issue is only about client AUTH.